### PR TITLE
Remove deprecated BIPMapping pass

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit removes the deprecated BIPMapping pass. It was deprecated in qiskit-terra 0.24.0 and the minimum deprecation period has elapsed for the Qiskit 0.45.0 release. This pass has been moved to an external plugin that can be installed separately and integrated into the transpiler more cleanly than a single pass and also separates the optional dependency on the proprietary CPLEX into a separate package.

### CODE REVIEW

File: releasenotes/notes/remove-deprecated-bip-mapper-e1206c8f905502dd.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    The deprecated transpiler routing pass, ``BIPMapping`` has been removed.
+    It was marked as deprecated in the Qiskit 0.43.0 release. It has been
+    replaced by an external plugin package: ``qiskit-bip-mapper``. Details for
+    this new package can be found at the package's github repository:
+
+        https://github.com/qiskit-community/qiskit-bip-mapper
+
+    The pass was made into a separate plugin package for two reasons, first
+    the dependency on CPLEX makes it harder to use and secondly the plugin
+    packge more cleanly integrates with :func:`~.transpile`. The optional
```
Comment (kevinhartman) [inline releasenotes/notes/remove-deprecated-bip-mapper-e1206c8f905502dd.yaml:13]:
```suggestion
    package more cleanly integrates with :func:`~.transpile`. The optional
```

File: releasenotes/notes/remove-deprecated-bip-mapper-e1206c8f905502dd.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    extra ``bip-mapper`` to install the ``cplex`` and ``docplex`` to support
+    this pass has removed as nothing in Qiskit optionally requires it anymore.
```
Comment (kevinhartman) [inline releasenotes/notes/remove-deprecated-bip-mapper-e1206c8f905502dd.yaml:15]:
```suggestion
    this pass has been removed as nothing in Qiskit optionally requires it anymore.
```

**Review by jakelishman — COMMENTED** [review]
Can you update the docs and install hints around `HAS_CPLEX` and `HAS_DOCPLEX` to comment that they're no longer used by Qiskit, and update their `install` hints to remove the reference to the (already) non-existent extra `bip-mapper`?

**Comment by mtreinish** [comment]
I updated the docs in: https://github.com/Qiskit/qiskit/pull/10526/commits/83a0d28cfcfe046017760b3aa3357de5fa03b688

**Comment by jakelishman** [comment]
There's also the `install='pip install qiskit-terra[bip-mapper]'` lines in the actual definitions of those two objects (lines 249 and 255 of `qiskit/utils/optionals.py`) if you want to get those in this PR.  It's pre-existing though (the optional was removed already), so we can get it in a follow-up if you prefer.

**Comment by mtreinish** [comment]
I updated them in: https://github.com/Qiskit/qiskit/pull/10526/commits/2a72879bb9a0348be1456331c7d8892aa964babd it makes sense to do it here even if it was pre-existing. I felt a bit odd about the cplex one though, because `pip install cplex` is installing a "free" demo version that is very limited. The real way to install cplex is to buy it from IBM and install the full package which includes the python extension without limitations.

**Review by jakelishman — APPROVED** [review]
Yeah, the CPLEX situation was always a bit of a problem, which is in part why we've walked the path that's led us to this PR overall haha.

The actual `install` command in the checkers isn't really going to matter because nobody will see it any more, but at least now it's not obviously incorrect.

**Pull request merged**